tests/federation/transport/test_knocking.py

```diff
@@ -23,10 +23,14 @@
 from synapse.types import RoomAlias
 
 from tests.test_utils import event_injection
-from tests.unittest import FederatingHomeserverTestCase, TestCase
+from tests.unittest import (
+    FederatingHomeserverTestCase,
+    HomeserverTestCaseProtocol,
+    TestCase,
+)
 
 
-class KnockingStrippedStateEventHelperMixin(TestCase):
+class KnockingStrippedStateEventHelperMixin(TestCase, HomeserverTestCaseProtocol):
```

Comment:
I think you can make this sub-class `HomeserverTestCase` instead and not need the protocol.

Reply:
Ah, indeed you can! Silly mypy docs.